### PR TITLE
Clarify when keys are created

### LINKED ISSUE #15792: Docker Content Trust documentation - creating keys
The [Content trust in Docker](https://docs.docker.com/security/trust/content_trust/) docs state (my emphasis on creating):

> Make sure you read [Manage keys for content trust](https://docs.docker.com/security/trust/trust_key_mng) information for details on **creating**, securing, and backing up your keys.

But there's nothing about creating keys on the [Manage keys for content trust](https://docs.docker.com/security/trust/trust_key_mng) page because they're actually created on the first push as per:

> When you push your first tagged image with content trust enabled, the docker client recognizes this is your first push and:
>* alerts you that it will create a new offline key
>* requests a passphrase for the key
>* generates an offline key in the ~/.docker/trust directory
>* generates a tagging key for in the ~/.docker/trust directory

It would be beneficial if the key creation process was explained outside of the **Push trusted content** section.

Documentation issue - so not copying in version/info etc.

### PR DESCRIPTION
**Pull request by cpswan**
Fixes #15792

Signed off by: Chris Swan chris.swan@iee.org

### CODE REVIEW

**Comment by endophage** [comment]
LGTM

**Comment by jfrazelle** [comment]
the commit is still signed incorrectly

**Comment by endophage** [comment]
@jfrazelle needs email rather than handle???

**Comment by jfrazelle** [comment]
I mean i just dont think it was done via the cli it looks off... and the regex isnt working :(

**Comment by calavera** [comment]
Please, do not move this forward until the commit is properly signed, with the email.

**Comment by cpswan** [comment]
Sorry @jfrazelle and @calvera this is my first PR to Docker, and despite trying my hardest to follow the excellent guidance from @moxiegirl I've obviously still managed to mess it up

If I follow Gordon the Turtle's guidance then I see:

```
git commit --amend -s --no-edit
[fixes-15792 10d3487] Clarify when keys are created
 Author: Chris Swan <chris.swan@iee.org>
 1 file changed, 3 insertions(+), 2 deletions(-)
```

Then I did `git push -f` and everything seemed to look OK on Github. So what do I need to do now to make this work?

PS since this is a couple of lines in docs I don't see how this could have anything to do with `Jenkins build Windows-PRs 12700 has failed`, but I'm willing to suspend disbelief on that.

**Comment by cpswan** [comment]
I think I got it - having only set my `user.email` locally for the `docker-fork` it wasn't applied when I cloned to another repo per the @GordonTheTurtle guidance (and I'd messed up my original commit by missing the `-s` flag).

**Comment by moxiegirl** [comment]
@cpswan Looks like you are almost there. Your signature went through but you need to rebase -- pull the latest material from docker/docker's master branch. This explains the rebase:

http://docs.docker.com/project/create-pr/#rebase-your-branch

File: docs/security/trust/content_trust.md [hunk 1]
```diff
@@ -104,8 +104,9 @@ content hash always succeeds as long as the hash exists:
 $ docker pull someimage@sha256:d149ab53f8718e987c3a3024bb8aa0e2caadf6c0328f1d9d850b2a2a67f2819a
 ```
 
-Trust for an image tag is managed through the use of signing keys. Docker's content
-trust makes use four different keys:
+Trust for an image tag is managed through the use of signing keys. A key set is
+created when an operation using content trust is first invoked. Docker's content
+trust makes use four different keys:
```
Comment (RichardScothern) [inline docs/security/trust/content_trust.md]: 'makes use of'

**Comment by cpswan** [comment]
@RichardScothern agreed that the the 'makes use of' needs to be fixed, but it's not an issue that's been introduced by this PR. If team Docker think it's something that should be done whilst we're here then I'm happy to do another (not just signing) update.

**Comment by cpswan** [comment]
I've fixed the nearby typo that @RichardScothern spotted

**Comment by moxiegirl** [comment]
@cpswan Thank you. LGTM

**Pull request merged**